compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -889,14 +895,14 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
 
                         // If all non-ZST fields have the same ABI, forward this ABI
                         if optimize && !field.is_zst() {
-                            // Normalize scalar_unit to the maximal valid range
+                            // Normalize scalar_unit to the maximal valid range and allow undef
```

Comment:
Now that `Scalar::Union` doesn't carry valid range information at all, the original portion of this comment seems kinda bogus. Maybe:

> Discard valid range information and allow undef